integration/fuse/src/main/java/alluxio/fuse/file/FuseFileInOrOutStream.java

```diff
@@ -158,14 +158,21 @@ public synchronized void truncate(long size) {
       mOutStream.get().truncate(size);
       return;
     }
-    if (size == getFileLength()) {
+    long currentSize = getFileLength();
+    if (size == currentSize) {
       return;
     }
-    if (size == 0) {
+    if (size == 0 || currentSize == 0) {
       AlluxioFuseUtils.deletePath(mFileSystem, mUri);
```

Comment:
are we sure there is no current fd opened for this path?

Reply:
This is an extreme case, like two threads:
one is opening one file, with fd1
then second one is opening it again, with fd2.

This deletion will directly delete the file, that will leave the fd1 dangling.

I am wondering if this scenario exists. This change looks good to me.

Reply:
I think this create() -> open() -> truncate() is basically creating a file. Your scenario is valid, but I don't think in real life there's workload accessing the file while creating it.

Reply:
From my understanding truncate is almost only used for testing purpose?